Sources/StreamChat/StateLayer/Chat.swift

```diff
@@ -518,7 +539,11 @@ public class Chat {
     ///
     /// - Throws: An error while communicating with the Stream API.
     /// - Returns: An array of pinned messages for the specified pagination.
-    public func loadPinnedMessages(for pagination: PinnedMessagesPagination? = nil, sort: [Sorting<PinnedMessagesSortingKey>] = [], limit: Int = .messagesPageSize) async throws -> [ChatMessage] {
+    public func loadPinnedMessages(
+        with pagination: PinnedMessagesPagination? = nil,
+        sort: [Sorting<PinnedMessagesSortingKey>] = [],
+        limit: Int = .messagesPageSize
+    ) async throws -> [ChatMessage] {
         let query = PinnedMessagesQuery(pageSize: limit, sorting: sort, pagination: pagination)
         return try await channelUpdater.loadPinnedMessages(in: cid, query: query)
     }
```

Comment:
`PinnedMessagesPagination` has sophisticated options which are not currently represented with convenience methods. I am not sure if it is needed at this point.
```swift
public enum PinnedMessagesPagination: Hashable {
    /// When used, the backend returns messages around the message with the given id.
    case aroundMessage(_ messageId: MessageId)

    /// When used, the backend returns messages pinned earlier then the message with the given id.
    ///
    /// When `inclusive == true` the results include the message with the given id.
    case before(_ messageId: MessageId, inclusive: Bool)

    /// When used, the backend returns messages pinned earlier then the message with the given id.
    ///
    /// When `inclusive == true` the results include the message with the given id.
    case after(_ messageId: MessageId, inclusive: Bool)

    /// When used, the backend returns messages pinned at around the given timestamp.
    case aroundTimestamp(_ timestamp: Date)

    /// When used, the backend returns messages pinned earlier than the given timestamp.
    ///
    /// When `inclusive == true` the results include the message pinned at the given timestamp.
    case earlier(_ timestamp: Date, inclusive: Bool)

    /// When used, the backend returns messages pinned later than the given timestamp.
    ///
    /// When `inclusive == true` the results include the message pinned at the given timestamp.
    case later(_ timestamp: Date, inclusive: Bool)
}
```

Reply:
Best to keep it as it is now - no new convenience methods.